lib/safira/contest/contest.ex

```diff
@@ -125,9 +126,17 @@ defmodule Safira.Contest do
   end
 
   def create_redeem(attrs \\ %{}) do
-    %Redeem{}
-    |> Redeem.changeset(attrs)
-    |> Repo.insert()
+    Multi.new()
+    |> Multi.insert(:redeem, Redeem.changeset(%Redeem{}, attrs))
+    |> Multi.update(:attendee, fn %{redeem: redeem} ->
+      redeem = Repo.preload(redeem, [:badge, :attendee])
+
+      Changeset.change(redeem.attendee,
+        token_balance: redeem.attendee.token_balance + redeem.badge.tokens
+      )
+    end)
+    |> Repo.transaction()
+    |> (fn x -> {elem(x, 0), elem(x, 1) |> Map.get(:redeem)} end).()
   end
```

Comment:
This logic works perfectly when a badge is being redeemed the first time. The problem is that that when a badge is being redeemed a second time to the same attendee (which normally results in a response stating that the badge has already been redeemed) it will cause an internal server error. 

In picture, error causes when trying to redeem badge with id:2 to the same attendee twice 
![image](https://user-images.githubusercontent.com/46303606/107874702-327a2880-6eb3-11eb-9bf4-5428dad2ef57.png)

I'm no Multi expert so I can't sugest what the correct change should be, but maybe @joaop21 @ZePedroResende @0urobor0s can.

Reply:
My bet would be that maybe line 131 shouldn't assume that the insert was succesful.

Reply:
It's not a multi problem. It's the line 139 that assumes a correct result every time.

Reply:
With the new commit i think this is solved.